tests/network/services.go

```diff
@@ -291,33 +256,47 @@ var _ = SIGDescribe("Services", func() {
 				})
 
 				By("checking connectivity the exposed service")
-				var err error
+				tcpJob, err := createConnectivityJobToService(serviceName, inboundVMI.Namespace, servicePort)
+				Expect(err).NotTo(HaveOccurred())
+				DeferCleanup(func() {
+					err := virtClient.BatchV1().Jobs(tcpJob.Namespace).Delete(context.Background(), tcpJob.Name, k8smetav1.DeleteOptions{})
+					Expect(err).To(SatisfyAny(
+						Not(HaveOccurred()),
+						MatchError(errors.IsNotFound, "does not exist"),
+					), cleaningK8sv1JobShouldSucceed)
+				})
 
-				jobCleanup, err = assertConnectivityToService(serviceName, inboundVMI.Namespace, servicePort)
-				Expect(err).NotTo(HaveOccurred(), expectConnectivityToExposedService)
+				Expect(job.WaitForJobToSucceed(tcpJob, 90*time.Second)).To(Succeed(), expectConnectivityToExposedService)
 			},
 				Entry("when the service is exposed by an IPv4 address.", k8sv1.IPv4Protocol),
 				Entry("when the service is exposed by an IPv6 address.", k8sv1.IPv6Protocol),
 			)
 		})
 
 		Context("*without* a service matching the vmi exposed", func() {
-			var jobCleanup func() error
-			var serviceName string
-
-			AfterEach(func() {
-				Expect(jobCleanup).NotTo(BeNil(), cleaningK8sv1JobFuncShouldExist)
-				Expect(jobCleanup()).To(Succeed(), cleaningK8sv1JobShouldSucceed)
-				jobCleanup = nil
-			})
-
 			It("should fail to reach the vmi", func() {
-				var err error
-				serviceName = "missingservice"
+				tcpJob, err := createConnectivityJobToService("missingservice", inboundVMI.Namespace, servicePort)
+				Expect(err).NotTo(HaveOccurred())
+
+				DeferCleanup(func() {
+					err := virtClient.BatchV1().Jobs(tcpJob.Namespace).Delete(context.Background(), tcpJob.Name, k8smetav1.DeleteOptions{})
+					Expect(err).To(SatisfyAny(
+						Not(HaveOccurred()),
+						MatchError(errors.IsNotFound, "does not exist"),
+					))
+				})
 
-				jobCleanup, err = assertNoConnectivityToService(serviceName, inboundVMI.Namespace, servicePort)
+				err = job.WaitForJobToFail(tcpJob, 89*time.Second)
 				Expect(err).NotTo(HaveOccurred(), "connectivity is *not* expected, since there isn't an exposed service")
 			})
 		})
 	})
 })
+
+func createConnectivityJobToService(serviceName string, namespace string, servicePort int) (*batchv1.Job, error) {
```

Comment:
Not sure about the English, but how about `createServiceConnectivityJob` ?

Reply:
Done